Blockbase: add additional vertical padding

mikachan opened the pull request:
<!-- Thanks for contributing to our free themes! Please provide as much information as possible with your Pull Request by filling out the following - this helps make reviewing much quicker! -->

#### Changes proposed in this Pull Request:
This change adds an additional `40px` of padding in between the header and main content on Blockbase. It also adds the same additional amount of padding in between each post in a query block.

I've made sure that the post title position is identical when navigating between the topmost post on an archive page and the single post page.

#### Related issue(s):
Closes #3964

mikachan committed:
Add custom vertical padding to theme.json

mikachan committed:
Add custom vertical padding to header

mikachan committed:
Add custom vertical padding to post template

mikachan committed:
Add new compiled ponyfill.css

mikachan committed:
Add newline at end of post-template scss file

scruffian commented on blockbase/sass/blocks/_post-template.scss:
```diff
@@ -0,0 +1,9 @@
+.wp-block-post-template {
+    li {
+        padding-bottom: var(--wp--custom--padding--vertical);
```
Will this also add padding bottom to list items inside posts that are loaded in the post template block?

mikachan replied:
Yes it probably would, I'll make it more specific!

scruffian commented:
Do you think adding a separator block to the template would work? The advantage of that approach is that users can customize it...

pbking commented on blockbase/theme.json:
```diff
@@ -203,6 +203,9 @@
 				"horizontal": "30px",
 				"vertical": "30px"
 			},
+			"padding": {
```
I hesitate to add a custom value for this to blockbase for this. Would it be possible to do this just with block styles rather than a custom variable + CSS?  Perhaps just adding a top padding of `40px` (or whatever is appropriate) to the ["core/post-title" block](https://github.com/Automattic/themes/blob/trunk/blockbase/theme.json#L386) would have the intended effect?

mikachan replied:
Yeah, applying a top padding to the 'core/post-title' block should have the same effect. I nearly went a similar route (although still with CSS..), but I added a reuseable variable as the same additional padding needed to be added to the posts in a query block. We could just do the same for these as well though - add the padding to the block in the `theme.json` (I'm guessing so anyway, without looking into it yet!)

mikachan committed:
Make post template <li> styling more specific

mikachan commented:
> Do you think adding a separator block to the template would work? The advantage of that approach is that users can customize it...

Yeah, that definitely sounds like a good option. That would make it easier to apply the same spacing in other places, without needing to change CSS or the `theme.json` each time (I think?) I'll look at this today.

mikachan committed:
Remove custom CSS for vertical padding

mikachan committed:
Add bottom padding to header template part

mikachan committed:
Add spacer of 40px to query loop

mikachan commented:
I've changed how this works based on your feedback and some help from @MaggieCabrera 🙌

Instead of using custom variables and additional CSS, I've tried to use the block editor as much as possible. This means fewer changes to the code and users can now customise both of these padding changes.

For the padding between the header and the titles, I've added `40px` of padding to the bottom of the header `wp:group` block.

For the padding underneath each post in a query list, I've added a `wp:spacer` block with a height of `40px`.

pbking commented on blockbase/block-templates/index.html:
```diff
@@ -6,6 +6,7 @@
 	<!-- wp:post-title {"isLink":true} /-->
 	<!-- wp:post-featured-image {"isLink":true} /-->
 	<!-- wp:post-excerpt /-->
+	<!-- wp:spacer {"height":40} -->
```
The spacer block needs to have the HTML stuff inside of it and get closed off to work properly.  I usually just instantiate the block in the block editor and copy/paste 'cause I usually forget how it goes. :P

```
	<!-- wp:spacer {"height":40} -->
	<div style="height:40px" aria-hidden="true" class="wp-block-spacer"></div>
	<!-- /wp:spacer -->

pbking commented on blockbase/block-templates/index.html:
```diff
@@ -6,6 +6,7 @@
 	<!-- wp:post-title {"isLink":true} /-->
```
KIND of unrelated to the change, but in order to get multiple items to show up to evaluate the spacer below I had to change this `<!-- wp:query-loop -->` to `<!-- wp:post-template -->` to get multiple items per-page.  This was a recent change to Gutenberg and would fit nicely into this change I think.

See [this change](https://github.com/Automattic/themes/pull/4107) for the same change in quadrat's templates.

mikachan replied:
Ahh I did this see when I was trying to figure out the query loop changes. I wasn't sure which one was correct so I went with the current one. This makes sense, thanks for explaining!

pbking commented:
Dig the padding and spacer changes better for sure.  A couple of things to fix up before we bring it in.

mikachan committed:
Add HTML content and closer to wp:spacer block

mikachan committed:
Change wp:query-loop to wp:post-template

mikachan commented:
This is now ready for another review. Thanks!

scruffian reviewed (approved):
This looks great, thanks.

A couple of things I also considered...

Should we add the padding to the template part, rather than the block. This seems easier for a user to edit as selecting the group block inside the template part is tricky. However we want this padding on every page, so I believe the group block is the right place for it, as you have done.

I also considered putting the spacer above the post, then we wouldn't need to add it to the header at all, but this would then also mean adding the same spacer on the singular template,  and if a user want to change it they'd need to update both places, so again, this solution is better!
